sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -119,6 +119,9 @@ class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
     @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution) extends Serializable {
 
+  // Note for Spark contributors: if adding or updating any action in `DataFrame`, please make sure
```

Comment:
> @zsxwing This looks great. One high level question: why are there so few dataframe operations with withNewExecutionId? Are you wrapping only those that will trigger an action? If so you should clarify that in the java doc.

@andrewor14 I added the comment here because I think it should only be visible to contributors.